src/Framework/PaymentGateways/Routes/GatewayRoute.php

```diff
@@ -86,4 +102,22 @@ private function isValidListener()
     {
         return isset($_GET['give-listener']) && $_GET['give-listener'] === 'give-gateway';
     }
+
+    /**
+     * Validate signature using nonces
+     *
+     * @unreleased
+     *
+     * @param  string  $routeSignature
+     * @param  GatewayRouteData  $data
+     * @return void
+     */
+    private function validateSignature($routeSignature, GatewayRouteData $data)
+    {
+        $action = RouteSignature::make($data->gatewayId, $data->gatewayMethod, $data->queryParams);
+
+        if (!wp_verify_nonce($routeSignature, $action->toString())) {
+            wp_redirect(response('Forbidden', 403));
```

Comment:
@jonwaldstein 

- I think this will cause an error because [`wp_redirect`](https://developer.wordpress.org/reference/functions/wp_redirect/) expects string param instead of `Response|ResponseFactory`.
- Always use `exit` after `wp_redirect`

Reply:
Figured it out, I needed to use wp_die for this